.github/workflows/ci.yml

```diff
@@ -0,0 +1,153 @@
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04, ubuntu-24.04, macos-14]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Deps
+        run: |
+          rustup default ${{ env.RUST_TOOLCHAIN }}
+          rustup component add rustc-dev rust-src
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: "true"
+          prefix-key: "${{ env.CACHE_VERSION }}-${{ matrix.os }}"
+      - run: cargo test --locked
+      - name: Install mir-json binaries
+        run: cargo install --locked
+      - name: Translate modified Rust standard libraries
+        run: mir-json-translate-libs
+      # NB: These binary distributions will not work unless you have the
+      # appropriate Rust toolchain installed beforehand.
+      - name: Extract executables to binary distribution
+        shell: bash
+        run: .github/ci.sh setup_dist_bins
+      - name: Compress binary distribution
+        shell: bash
+        run: |
+          NAME="mir-json-${{ matrix.os }}-${{ runner.arch }}"
+          echo "NAME=$NAME" >> $GITHUB_ENV
+          .github/ci.sh zip_dist $NAME
+      - name: Upload binary distribution
+        uses: actions/upload-artifact@v4
+        if: github.event.pull_request.head.repo.fork == false && github.repository_owner == 'GaloisInc'
+        with:
+          name: ${{ env.NAME }}
+          path: "deps/mir-json/${{ env.NAME }}.tar.gz*"
+          if-no-files-found: error
+  package-distribution:
+    runs-on: ${{ matrix.os }}
+    needs: [build-haskell, build-mir-json]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04, ubuntu-24.04, macos-14]
+    steps:
+      - name: Download Haskell binaries
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ matrix.os }}-${{ runner.arch }}-bins
+          path: dist/bin
+      - name: Download mir-json binaries
+        uses: actions/download-artifact@v4
+        with:
+          name: mir-json-${{ matrix.os }}-${{ runner.arch }}
+      - name: Download solver binaries
+        run: |
+          mkdir -p solver-bin
+          cd solver-bin
+          curl -o bins.zip -sL "https://github.com/GaloisInc/what4-solvers/releases/download/${{ env.SOLVER_PKG_VERSION }}/${{ matrix.os }}-${{ runner.arch }}-bin.zip"
+          unzip -o bins.zip
+          rm bins.zip
+          cd ..
+      - name: Build archive
+        run: |
+          cp -v solver-bin/* dist/bin
+          tar xzvf mir-json-${{ matrix.os }}-${{ runner.arch }}.tar.gz
+          mv -v mir-json-${{ matrix.os }}-${{ runner.arch }}/bin dist/rust-bin
+          mv -v mir-json-${{ matrix.os }}-${{ runner.arch }}/rlibs dist/rlibs
+          chmod a+x dist/bin/*
+      - name: Upload final artifact
+        uses: actions/upload-artifact@v4
+        with:
+          path: dist
+          name: saw-suite-${{ matrix.os }}-${{ runner.arch }}
```

Comment:
I downloaded one of the uploaded artifacts and unzipped it, but I was surprised to discover that instead of unzipping the contents to a single directory (named `saw-suite-*` or some such), it dumped the raw contents directly into my working directory (e.g., it dumped raw `bin` and `rust-bins` directories). This seems undesirable. Can you put everything under a single `saw-suite-*` directory instead?